src/game.py

```diff
@@ -9,28 +9,40 @@ def __init__(self, cli=Cli(), board=Board(), players=[Player(token="X"),
         self.cli = cli
         self.board = board
         self.players = players
+        self.exit = False
 
     def start(self):
         self.cli.welcome()
         self.play()
 
     def play(self):
         self.cli.display_board(self.board)
-        while not self.board.full():
+        while not self.game_is_over():
             self.turn()
+        self.handle_exit()
 
-        self.cli.goodbye()
-    
-    def turn(self, error=False):
-        if error: self.cli.invalid_move()
-
-        player_choice = self.cli.prompt_player_turn(self.current_player())
-        move = self.input_to_index(player_choice)
-        if self.board.valid_move(move):
+    def handle_exit(self):
+        if self.exit:
+            self.cli.handle_exit()
+        else:
+            self.cli.goodbye()
```

Comment:
This method reads strangely. If exit, exit. Else goodbye. As a reader it isn't clear what the diff between exit and goodbye are. Might want to consider using more descriptive names.